gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/DagActionStoreChangeMonitor.java

```diff
@@ -146,15 +148,26 @@ protected void initializeMonitor() {
       throw new RuntimeException(String.format("Unable to retrieve dagActions from the dagActionStore while "
           + "initializing the %s", DagActionStoreChangeMonitor.class.getCanonicalName()), e);
     }
-    // TODO: make this multi-threaded to add parallelism
+    final ExecutorService executorService = Executors.newFixedThreadPool(ConfigUtils.getInt(this.config, ConfigurationKeys.DAG_ACTION_STORE_MONITOR_EXECUTOR_THREADS, 5));
+
     for (DagActionStore.DagAction action : dagActions) {
       try {
-        handleDagAction(action, true);
+        executorService.submit(()->handleDagAction(action, true));
```

Comment:
`handleDagAction` throughs RuntimeException. `submit` will swallow it.
You can use UncaughtExceptionHandler

Reply:
addressed